knox/knox.sh

```diff
@@ -128,8 +160,31 @@ function initialize_crontab() {
 function install() {
   local master_name
   master_name=$(/usr/share/google/get_metadata_value attributes/dataproc-master)
-  if [[ "${master_name}" == $(hostname) ]]; then
+  if [[ "${DATAPROC_IMAGE_VERSION}" == "2.2" ]]; then
```

Comment:
Please write double conditionals as

`if [[ "${DATAPROC_IMAGE_VERSION}" == "2.2" ]] && [[ "${master_name}" == $(hostname -s) ]] ; then`

rather than making the only contents of an if block another if block